test/src/com/connectsdk/service/DIALServiceTest.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * DIALServiceTest
+ * Connect SDK
+ *
+ * Copyright (c) 2015 LG Electronics.
+ * Created by Oleksii Frolov on 06 Aug 2015
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.connectsdk.service;
+
+import com.connectsdk.service.capability.Launcher;
+import com.connectsdk.service.command.ServiceCommand;
+import com.connectsdk.service.config.ServiceConfig;
+import com.connectsdk.service.config.ServiceDescription;
+
+import junit.framework.Assert;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.annotation.Config;
+
+@RunWith(RobolectricTestRunner.class)
+@Config(manifest=Config.NONE)
+public class DIALServiceTest {
+
+    private static final String APPLICATION_URL = "http://applicationurl";
+
+    private StubDialService service;
+    private ServiceDescription serviceDescription;
+    private ServiceConfig serviceConfig;
+
+    class StubDialService extends DIALService {
```

Comment:
I think it's better to directly set a `ServiceCommandProcessor` mock object (https://github.com/ConnectSDK/Connect-SDK-Android-Core/pull/66/files#diff-df781b69f87461dd3812a64d263f7a75L141) to verify commands instead of subclassing&overriding or partially mocking a real object. That's a different responsibility anyway, and should ideally be extracted.